### PR TITLE
Extendable typings

### PR DESCRIPTION
**Pull request by ablok**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

Proposal for adding types to frameworks/services etc...
Please have a look at the proposed changes here and let me know what you think. It does 2 things:
- The current typings make no difference in the `suite` and `test` description, regardless of framework. Actually, they are currently describing the mocha output. However, when running jasmine, there is different output available. With this PR all you would have to do is reference the framework package in your `tsconfig.json` and it would give you the correct typings for `suite` and `test` AND give you global typings for either mocha or jasmine.
- As an example I also added typings in this way for the `selenium-standalone-service`. To get the correct typings and allowed extra properties to the `config` object, all you need to do is install selenium-standalone-service and add a reference to it in the `tsconfig.json`.

Look at the `typescript.md` file for a description/example. This is just a proposal, but one that would work pretty nicely I think. Ideally, all wdio packages would come with typings files like this.

Please let me know what you think about this.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: docs/TypeScript.md [hunk 1]
```diff
@@ -41,3 +41,53 @@ and your `tsconfig.json` needs to look like:
     ]
 }
 ```
+
+You can even use a typed configuration if you desire.
+Al you have to do is create a plain js config file that registers typescript and requires the typed config:
```
Comment (christian-bromann) [inline docs/TypeScript.md]:
```suggestion
All you have to do is create a plain js config file that registers typescript and requires the typed config:
```

File: docs/TypeScript.md [hunk 2]
```diff
@@ -41,3 +41,53 @@ and your `tsconfig.json` needs to look like:
...
+
+```javascript
+require("ts-node/register")
+module.exports = require("wdio.conf.ts")
+```
+
+
```
Comment (christian-bromann) [inline docs/TypeScript.md]:
```suggestion
```

File: docs/TypeScript.md [hunk 3]
```diff
@@ -41,3 +41,53 @@ and your `tsconfig.json` needs to look like:
...
+```typescript
+const config: WebdriverIO.Config = {
+    // Put your webdriverio configuration here
+}
+
+export { config }
+```
+
+
```
Comment (christian-bromann) [inline docs/TypeScript.md]:
```suggestion
```

File: docs/TypeScript.md [hunk 4]
```diff
@@ -41,3 +41,53 @@ and your `tsconfig.json` needs to look like:
...
+        },
+        "types": ["node", "webdriverio", "@wdio/mocha-framework"]
+    },
+    "include": [
+        "./src/**/*.ts"
+    ]
+}
+```
+
+
```
Comment (christian-bromann) [inline docs/TypeScript.md]:
```suggestion
```

File: docs/TypeScript.md [hunk 5]
```diff
@@ -41,3 +41,53 @@ and your `tsconfig.json` needs to look like:
...
+Instead of having all type definitions globally available, you can also import only the typings that you need like this:
+
+```typescript
+/*
+* These import the type definition for the `test` and `suite` variables that are available in
+* the beforeTest, afterTest, beforeSuite and afterSuite hooks.
+*/
+import { Suite, Test } from "mocha-framework"
```
Comment (christian-bromann) [inline docs/TypeScript.md]:
```suggestion
import { Suite, Test } from "@wdio/mocha-framework"  
```

**Comment by ablok** [comment]
@christian-bromann I processed the review comments.
Do you agree with this approach? If so, I could try to add some more typings to other services in some more PR's.

**Comment by christian-bromann** [comment]
> Do you agree with this approach? If so, I could try to add some more typings to other services in some more PR's.

Yes, I mean I am not a typescript expert but I think it makes sense to add sense and I don't worry about adding a small typings package to the subpackages.

**Pull request merged**